src/scheduler/node_info.c

```diff
@@ -4623,6 +4620,22 @@ reorder_nodes(node_info **nodes, resource_resv *resresv)
 	node_array[0] = NULL;
 	nptr = node_array;
 
+	if (resresv->is_resv && resresv->resv && resresv->resv->check_alternate_nodes) {
```

Comment:
Check pointers against NULL, so if (ptr != NULL) not if (ptr)

Reply:
done!